src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2021,15 +2027,19 @@ private MemberDeclarationSyntax ParseMemberDeclarationOrStatementCore(SyntaxKind
                 }
             }
 
-            var modifiers = _pool.Allocate();
-
             var saveTermState = _termState;
-            var startPoint = this.GetResetPoint();
+
+            var attributes = this.ParseAttributeDeclarations();
+            bool haveAttributes = (attributes.Count > 0);
+
+            var afterAttributesPoint = this.GetResetPoint();
+
+            var modifiers = _pool.Allocate();
 
             try
             {
-                var attributes = this.ParseAttributeDeclarations();
-                bool haveAttributes = (attributes.Count > 0);
+                TerminatorState saveTerm;
+                bool wasInAsync;
```

Comment:
>bool wasInAsync [](start = 16, length = 15)

Please consider removing this local and keeping the individual locals in each scope, for clarity.